Competition2017/tests/competition/subsystems/shooter/SideShooterTest.java

```diff
@@ -0,0 +1,92 @@
+package competition.subsystems.shooter;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import competition.subsystems.shooter.commands.ContinueShooterCommand;
+import competition.subsystems.shooter.commands.SetShooterSpeedCommand;
+import competition.subsystems.shooter.commands.StepShooterPowerCommand;
+import competition.subsystems.shooter.commands.StopShooterCommand;
+
+public class SideShooterTest extends ShooterTestBase {
+    
+     SideShooterSubsystem leftShooter;
+     SideShooterSubsystem rightShooter;
+    
+    @Before
+    @Override
+    public void setup() {
+        super.setup();
+        leftShooter = shooter.getLeftShooter();
+        rightShooter = shooter.getRightShooter();
+        
+    }
+    
+    @Test
+    public void testContinueShooterCommand(){
+        ContinueShooterCommand cmd = injector.getInstance(ContinueShooterCommand.class);
+        
+        cmd.setSide((leftShooter));
+        cmd.initialize();
+        cmd.execute();
+        cmd.isFinished();
+        
+        cmd.setSide(rightShooter);
+        cmd.initialize();
+        cmd.execute();
+        cmd.isFinished();
+        
+         verifyShooterPowers(shooter.getLeftShooter().getLauncherPower(), shooter.getRightShooter().getLauncherPower());;
+    }
+    
+    @Test
+    public void testStepShooterPowerCommand(){
+        StepShooterPowerCommand sspc = injector.getInstance(StepShooterPowerCommand.class);
+        sspc.setSide(leftShooter);
+        
+        double leftTemp = shooter.getLeftShooter().getLauncherPower();
+        
+        sspc.initialize();
+        sspc.execute();
+        sspc.isFinished();
+        
+        verifyShooterPowers(leftTemp + leftShooter.getPowerStep(), 0);
+    }
+    
+    @Test 
+    public void testStopShooterCommand(){
+        StopShooterCommand stop = injector.getInstance(StopShooterCommand.class);
+        stop.setSide(leftShooter);
+        
+        stop.initialize();
+        stop.execute();
+        stop.isFinished();
+        
+        verifyShooterPowers(0,0);
+        
+    }
+    
+    @Test
+    public void testSetShooterSpeedCommand(){
+        SetShooterSpeedCommand sssc = injector.getInstance(SetShooterSpeedCommand.class);
+        sssc.setSide(leftShooter);
+        
+        sssc.initialize();
+        sssc.execute();
+        sssc.isFinished();
+        
+        sssc.setSide(rightShooter);
```

Comment:
I think it would be better to create a second SetShooterSpeedCommand rather than calling setSide twice on the same one, under the covers calling it twice will set the `requires()` twice which would have unintended consequences on the robot